src/lib/ui/RadioField/RadioField.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useContext } from 'react'
+
+import { Radio } from 'lib/ui/RadioField/Radio.js'
+
+export interface RadioFieldProps<Value extends string> {
+  name: string
+  value: Value
+  onChange: (value: Value) => void
+  options: Array<RadioOption<Value>>
+}
+
+export interface RadioOption<Value> {
+  label: string
+  value: Value
+}
+
+interface RadioFieldContextProps<Value extends string> {
+  value: Value
+  onChange: (value: Value) => void
+  name: string
+}
+
+const RadioFieldContext = createContext<
+  RadioFieldContextProps<any> | undefined
```

Comment:
Not that I'm aware of, but this `any` might be ok, because:

- The generic is be specified via `useRadioField`.
- The `RadioFieldContext` isn't actually exposed (exported). The `any` is self-contained.
- The single call-site, inside `<Radio/>` forwards the type along:

```ts

export function Radio<Value extends string>({
  value,
  label,
}: RadioProps<Value>): JSX.Element {
  const { value: fieldValue, onChange, name } = useRadioField<Value>()
...
}
```

Ultimately, we still get strong typing where it matters, in the primary component: `RadioField`


![CleanShot 2023-07-20 at 15 08 37](https://github.com/seamapi/react/assets/11449462/1c284dfa-4ea0-4aa9-b225-e1f6c9a7e500)

We've passed in a union for the `value`, and...

![CleanShot 2023-07-20 at 15 09 33](https://github.com/seamapi/react/assets/11449462/2d023c9c-d1e1-40c7-b136-89f3076b8b0e)

The options are correctly inferred.